### ISSUE TITLE
What is difference between Height and Depth value?

### ISSUE THREAD

**Issue by LeVHoangduc** [rhett-chen/graspness_implementation#27]
At models/graspnet.py
At def pred_decode(end_points):
    batch_size = len(end_points['point_clouds'])
    grasp_preds = []
    for i in range(batch_size):
        grasp_center = end_points['xyz_graspable'][i].float()

        grasp_score = end_points['grasp_score_pred'][i].float()
        grasp_score = grasp_score.view(M_POINT, NUM_ANGLE*NUM_DEPTH)
        grasp_score, grasp_score_inds = torch.max(grasp_score, -1)  # [M_POINT]
        grasp_score = grasp_score.view(-1, 1)
        grasp_angle = (grasp_score_inds // NUM_DEPTH) * np.pi / 12
        grasp_depth = (grasp_score_inds % NUM_DEPTH + 1) * 0.01
        grasp_depth = grasp_depth.view(-1, 1)
        grasp_width = 1.2 * end_points['grasp_width_pred'][i] / 10.
        grasp_width = grasp_width.view(M_POINT, NUM_ANGLE*NUM_DEPTH)
        grasp_width = torch.gather(grasp_width, 1, grasp_score_inds.view(-1, 1))
        grasp_width = torch.clamp(grasp_width, min=0., max=GRASP_MAX_WIDTH)

        approaching = -end_points['grasp_top_view_xyz'][i].float()
        grasp_rot = batch_viewpoint_params_to_matrix(approaching, grasp_angle)
        grasp_rot = grasp_rot.view(M_POINT, 9)

        # merge preds
        grasp_height = 0.02 * torch.ones_like(grasp_score)
        obj_ids = -1 * torch.ones_like(grasp_score)
        grasp_preds.append(
            torch.cat([grasp_score, grasp_width, grasp_height, grasp_depth, grasp_rot, grasp_center, obj_ids], axis=-1))
    return grasp_preds

What do grasp height and grasp depth mean? do they both mean height and depth from camera to object?

**Comment by rhett-chen** [comment]
You can refer to [grasp format](https://graspnetapi.readthedocs.io/en/latest/grasp_format.html#d-grasp). The grasp height is a fixed value, and the grasp depth is the distance from grasp point (point on object surface) to the center of the two contact points.

**Comment by LeVHoangduc** [comment]
> You can refer to [grasp format](https://graspnetapi.readthedocs.io/en/latest/grasp_format.html#d-grasp). The grasp height is a fixed value, and the grasp depth is the distance from grasp point (point on object surface) to the center of the two contact points.

I followed this [paper ](https://openaccess.thecvf.com/content/ICCV2021/papers/Wang_Graspness_Discovery_in_Clutters_for_Fast_and_Accurate_Grasp_Detection_ICCV_2021_paper.pdf)and it says this " Cylinder-Grouping from Seed Points The locations and directions of cylinder spaces are determined by seed point coordinates and view vectors respectively. For each of the M point-view pairs, we group and sample K points from M seed points using the cylinder with fixed height d and radius r. After aligning the cylinder with gripper frame as [13], the point coordinates are normalized by cylinder radius and concatenated with feature vectors which are the sum of features output by graspable FPS and graspable PVS. The grouped point sets of size M × K × (3 + C) are called grasp candidates, where K stands for the number of sampled points in each group."
So, the fixed height stands for the height of **cylindrical region. ** for each seed points, right? and the unit of prediction result of Width, height and depth is "m" right?

**Comment by LeVHoangduc** [comment]
And I read this paper https://ieeexplore.ieee.org/abstract/document/9561409 . 
![image](https://github.com/rhett-chen/graspness_implementation/assets/104718152/710974af-4ac5-4584-a306-35303dda9ac6)
So, the height is the fixed value of gripper. So, why we need to predict the height value for what? Because I saw the output of model includes the height value.

**Comment by rhett-chen** [comment]
> And I read this paper https://ieeexplore.ieee.org/abstract/document/9561409 . ![image](https://private-user-images.githubusercontent.com/104718152/321678386-710974af-4ac5-4584-a306-35303dda9ac6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxMDUxNjYsIm5iZiI6MTcxMzEwNDg2NiwicGF0aCI6Ii8xMDQ3MTgxNTIvMzIxNjc4Mzg2LTcxMDk3NGFmLTRhYzUtNDU4NC1hMzA2LTM1MzAzZGRhOWFjNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNFQxNDI3NDZaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hZDAzMGRlNGZjOTVmYTlhMmI2ODc1YWI1NGJjZjI5ZDE3ZGE1ZGYzM2VjYTgyMDM1ZTNkOWIyMDhmMjMyNGI5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.vlR1On92I1953KazUFoSXnxHUaxxG9Ws4QKOSODUSBs) So, the height is the fixed value of gripper. So, why we need to predict the height value for what? Because I saw the output of model includes the height value.

The height value is used for collision detection.

**Comment by rhett-chen** [comment]
> > You can refer to [grasp format](https://graspnetapi.readthedocs.io/en/latest/grasp_format.html#d-grasp). The grasp height is a fixed value, and the grasp depth is the distance from grasp point (point on object surface) to the center of the two contact points.
> 
> I followed this [paper ](https://openaccess.thecvf.com/content/ICCV2021/papers/Wang_Graspness_Discovery_in_Clutters_for_Fast_and_Accurate_Grasp_Detection_ICCV_2021_paper.pdf)and it says this " Cylinder-Grouping from Seed Points The locations and directions of cylinder spaces are determined by seed point coordinates and view vectors respectively. For each of the M point-view pairs, we group and sample K points from M seed points using the cylinder with fixed height d and radius r. After aligning the cylinder with gripper frame as [13], the point coordinates are normalized by cylinder radius and concatenated with feature vectors which are the sum of features output by graspable FPS and graspable PVS. The grouped point sets of size M × K × (3 + C) are called grasp candidates, where K stands for the number of sampled points in each group." So, the fixed height stands for the height of **cylindrical region. ** for each seed points, right? and the unit of prediction result of Width, height and depth is "m" right?
No, the height and width in the gripper setting are not related to the cylindrical grouping here. You can use the graspnetAPI to visualize the gripper and understand what the width and height represent by changing them.

**Comment by rhett-chen** [comment]
> > You can refer to [grasp format](https://graspnetapi.readthedocs.io/en/latest/grasp_format.html#d-grasp). The grasp height is a fixed value, and the grasp depth is the distance from grasp point (point on object surface) to the center of the two contact points.
> 
> I followed this [paper ](https://openaccess.thecvf.com/content/ICCV2021/papers/Wang_Graspness_Discovery_in_Clutters_for_Fast_and_Accurate_Grasp_Detection_ICCV_2021_paper.pdf)and it says this " Cylinder-Grouping from Seed Points The locations and directions of cylinder spaces are determined by seed point coordinates and view vectors respectively. For each of the M point-view pairs, we group and sample K points from M seed points using the cylinder with fixed height d and radius r. After aligning the cylinder with gripper frame as [13], the point coordinates are normalized by cylinder radius and concatenated with feature vectors which are the sum of features output by graspable FPS and graspable PVS. The grouped point sets of size M × K × (3 + C) are called grasp candidates, where K stands for the number of sampled points in each group." So, the fixed height stands for the height of **cylindrical region. ** for each seed points, right? and the unit of prediction result of Width, height and depth is "m" right?

No, the height and width in the gripper setting are not related to the cylindrical grouping here. You can use the graspnetAPI to visualize the gripper and understand what the width and height represent by changing them.

**Comment by LeVHoangduc** [comment]
I'm very grateful for your clear and detailed explanation of the question. Thank you very much!
